qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -85,9 +85,23 @@ def run(self, dag):
         basic_instrs = ['measure', 'reset', 'barrier', 'snapshot']
 
         target_basis = set(self._target_basis).union(basic_instrs)
-        source_basis = {(node.op.name, node.op.num_qubits)
-                        for node in dag.op_nodes()}
-
+        # source_basis is a list here so that it can accurately be checked for calibrations.
+        source_basis = [(node.op.name, node.op.num_qubits)
+                        for node in dag.op_nodes()]
+
+        if dag.calibrations:
+            for node in dag.op_nodes():
+                gate = node.op.name
+                qubit_params = (tuple([node.qargs[0].index]), tuple(node.op.params))
+                num_qubits = node.op.num_qubits
+                if gate in dag.calibrations:
+                    gate_cals = dag.calibrations[gate]
+                    if qubit_params in gate_cals:
+                        if (gate, num_qubits) in source_basis:
+                            source_basis.remove((gate, num_qubits))
+
+        # type casting source_basis to set type.
+        source_basis = set(source_basis)
```

Comment:
```suggestion
        source_basis = set()
        for node in dag.op_nodes():
            name = node.op.name
            qubit_params = (tuple([node.qargs[0].index]), tuple(node.op.params))
            if (dag.calibrations and name in dag.calibrations
                and qubit_params in dag.calibrations[name]):
                pass
            else:
                source_basis.add((name, node.op.num_qubits))
```

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5080/commits/0ce6ad2f700d92fb0a6eed324fc7506e5e988b71